elastic/kibana
# [Security Solution][DE] Investigation fields telemetry

## Summary

Adds telemetry to determine the number of users utilizing `investigation_fields` in 8.10. We want to eventually deprecate use of this field as an array (in 8.10) in favor of the new object type. 

Utilizes the telemetry detections rule usage logic to add a new field `legacy_investigation_fields` - which is a total count of the number of rules utilizing the legacy `investigation_fields` (the field as an array). 

To test locally:
- Set the following type to `any` so that rule import is not blocked
    - File `x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.ts`, line `284`
- Import a rule using the legacy `investigation_fields` format:
```
{"id":"a08c-6796ed43cd60","updated_at":"2023-10-23T05:22:42.698Z","updated_by":"elastic","created_at":"2023-10-23T05:22:42.698Z","created_by":"elastic","name":"LEGACY","tags":[],"interval":"5m","enabled":true,"revision":0,"description":"mmmm","risk_score":21,"severity":"low","license":"","output_index":"","investigation_fields":["agent.name","client.address"],"author":[],"false_positives":[],"from":"now-360s","rule_id":"45","max_signals":100,"risk_score_mapping":[],"severity_mapping":[],"threat":[],"to":"now","references":[],"version":1,"exceptions_list":[],"immutable":false,"related_integrations":[],"required_fields":[],"setup":"","type":"query","language":"kuery","index":["apm-*-transaction*","auditbeat-*","endgame-*","filebeat-*","logs-*","packetbeat-*","traces-apm*","winlogbeat-*","-*elastic-cloud-logs-*"],"query":"*","filters":[],"actions":[]}
{"exported_count":1,"exported_rules_count":1,"missing_rules":[],"missing_rules_count":0,"exported_exception_list_count":0,"exported_exception_list_item_count":0,"missing_exception_list_item_count":0,"missing_exception_list_items":[],"missing_exception_lists":[],"missing_exception_lists_count":0,"exported_action_connector_count":0,"missing_action_connection_count":0,"missing_action_connections":[],"excluded_action_connection_count":0,"excluded_action_connections":[]}
```
- Check that the usage telemetry is updated in Advanced Settings --> Global settings tab --> scroll to the bottom and click on `cluster data`
<img width="1024" alt="Screenshot 2023-10-22 at 10 56 34 PM" src="https://github.com/elastic/kibana/assets/10927944/8f83ec1f-4b6e-4170-bf42-8f79988ca14c">

See (I had 4 in my instance):
<img width="363" alt="Screenshot 2023-10-22 at 10 49 21 PM" src="https://github.com/elastic/kibana/assets/10927944/84552928-6342-4616-9f73-b64f35e2f82c">
<img width="358" alt="Screenshot 2023-10-22 at 10 49 32 PM" src="https://github.com/elastic/kibana/assets/10927944/c7e1633a-aebd-486d-98be-9d2941bce1ba">
